src/Hal/Serializer/CollectionNormalizer.php

```diff
@@ -68,15 +69,18 @@ public function normalize($object, $format = null, array $context = [])
         $resourceClass = $this->resourceClassResolver->getResourceClass($object, $context['resource_class'] ?? null, true);
         $context = $this->initContext($resourceClass, $context);
         $parsed = IriHelper::parseIri($context['request_uri'] ?? '/', $this->pageParameterName);
-        $paginated = $isPaginator = $object instanceof PaginatorInterface;
 
-        $currentPage = $lastPage = $itemsPerPage = null;
-        if ($isPaginator) {
+        $currentPage = $lastPage = $itemsPerPage = $pageTotalItems = null;
+        if ($paginated = $isPaginator = $object instanceof PartialPaginatorInterface) {
+            if ($object instanceof PaginatorInterface) {
+                $paginated = 1. !== $lastPage = $object->getLastPage();
+            } else {
+                $lastPage = null;
```

Comment:
Is this one needed if we declared it above?

Reply:
Done.